### PR TITLE
Client movies

### PR DESCRIPTION
**Pull request by LisaGenner**
updates for 3.5 exercise

### CODE REVIEW

File: src/components/login-view/login-view.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    };
+
+    return (
+        <form>
+            <label>
+                Username:
+                <input type="text"
+                    value={username}
+                    onChange={(e) => setAbc(e.target.value)}
```
Comment (maskaravivek) [inline src/components/login-view/login-view.jsx]: It should be `setUsername`

File: src/components/login-view/login-view.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            });
+        if (response.ok) {
+            onLoggedIn(username);
+        } else {
+            alert("Login failed");
+        }
+    };
+
+    return (
+        <form>
```
Comment (maskaravivek) [inline src/components/login-view/login-view.jsx]: Add `onSubmit={handleSubmit}` to the form. 

```
<form onSubmit={handleSubmit}>
```

File: src/components/signup-view/signup-view.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+
+export const SignupView = () => {
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [email, setEmail] = useState("");
+    const [birthday, setBirthday] = useState("");
+    const handleSubmit = (event) => { };
```
Comment (maskaravivek) [inline src/components/signup-view/signup-view.jsx]: You need to implement the `handleSubmit` method similar to how you did for the `LoginView`.

File: src/components/login-view/login-view.jsx (hunk 1 above)
Comment (LisaGenner) [reply src/components/login-view/login-view.jsx]: forgot to change back after debugging

File: src/components/login-view/login-view.jsx (hunk 2 above)
Comment (LisaGenner) [reply src/components/login-view/login-view.jsx]: forgot to change back after debugging

File: src/components/signup-view/signup-view.jsx (hunk 3 above)
Comment (LisaGenner) [reply src/components/signup-view/signup-view.jsx]: updated code to include additional code

File: src/components/signup-view/signup-view.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        const data = {
+            Username: username,
+            Password: password,
+            Email: email,
+            Birthday: birthday
+        };
+
+        fetch("https://myflix-20778.herokuapp.com/login", {
+            method: "POST",
```
Comment (maskaravivek) [inline src/components/signup-view/signup-view.jsx:20]: Add the following line after `method: "POST"`

```
body: JSON.stringify(data)
```

File: src/components/signup-view/signup-view.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        event.preventDefault();
+
+        const data = {
+            Username: username,
+            Password: password,
+            Email: email,
+            Birthday: birthday
+        };
+
+        fetch("https://myflix-20778.herokuapp.com/login", {
```
Comment (maskaravivek) [inline src/components/signup-view/signup-view.jsx]: it should be `/users` instead of `/login`

File: src/components/login-view/login-view.jsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                    onChange={(e) => setEmail(e.target.value)}
+                />
+            </label>
+            <label>
+                Birthday:
+                <input type="date"
+                    value={birthday}
+                    onChange={(e) => setBirthday(e.target.value)}
+                />
+            </label>
```
Comment (maskaravivek) [inline src/components/login-view/login-view.jsx]: Delete line 64-77. Email id and birthday fields are not needed for login

File: src/components/login-view/login-view.jsx (hunk 2 above)
Comment (maskaravivek) [reply src/components/login-view/login-view.jsx]: You have added it incorrectly. It should be:

```
<form onSubmit={handleSubmit}>
```

Delete `line 48`.

File: src/components/login-view/login-view.jsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            <button type="submit">
+                Submit
+            </button>
+        </form>
+    );
```
Comment (maskaravivek) [inline src/components/login-view/login-view.jsx]: Delete the closing `)`

File: src/components/login-view/login-view.jsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                }
+            })
+            .catch((e) => {
+                alert("Something went wrong");
+            });
+        if (response.ok) {
+            onLoggedIn(username);
+        } else {
+            alert("Login failed");
+        }
```
Comment (maskaravivek) [inline src/components/login-view/login-view.jsx]: Delete these 4 lines

File: src/components/login-view/login-view.jsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+    const handleSubmit = (event) => {
+        //this prevents the default behavior of the form which is to reload the entire page
+        event.preventDefault();
+
+        const data = {
+            Username: username,
+            Password: password,
+            Email: email,
+            Birthday: birthday
```
Comment (maskaravivek) [inline src/components/login-view/login-view.jsx]: Delete these two lines

File: src/components/login-view/login-view.jsx [hunk 10]
```diff
@@ -0,0 +1,83 @@
+import { useState } from "react";
+
+export const LoginView = ({ onLoggedIn }) => {
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [email, setEmail] = useState("");
+    const [birthday, setBirthday] = useState("");
```
Comment (maskaravivek) [inline src/components/login-view/login-view.jsx]: Delete these two properties

File: src/components/signup-view/signup-view.jsx (hunk 4 above)
Comment (LisaGenner) [reply src/components/signup-view/signup-view.jsx:20]: updated

File: src/components/signup-view/signup-view.jsx (hunk 5 above)
Comment (LisaGenner) [reply src/components/signup-view/signup-view.jsx]: updated

File: src/components/login-view/login-view.jsx (hunk 6 above)
Comment (LisaGenner) [reply src/components/login-view/login-view.jsx]: updated

File: src/components/signup-view/signup-view.jsx (hunk 5 above)
Comment (maskaravivek) [reply src/components/signup-view/signup-view.jsx]: It still shows `/login` for me. Can you please check and update it to `/users`

**Comment by LisaGenner** [comment]
updates made to the file and many of them were already completed.  let me know if you see any issues seeing the updates

**Pull request closed without merging**